mbtiles/__init__.py

```diff
@@ -44,11 +45,17 @@ def process_tile(tile):
         *mercantile.ul(tile.x + 1, tile.y + 1, tile.z))
 
     kwds = base_kwds.copy()
-    kwds['transform'] = from_bounds(ulx, lry, lrx, uly, 256, 256)
+    kwds['transform'] = transform_from_bounds(ulx, lry, lrx, uly, 256, 256)
     src_nodata = kwds.pop('src_nodata', None)
     dst_nodata = kwds.pop('dst_nodata', None)
 
     with rasterio.open('/vsimem/tileimg', 'w', **kwds) as tmp:
+        (west, east), (south, north) = transform("EPSG:3857", src.crs,
+                                                 (ulx, lrx), (lry, uly))
+        tile_window = window_from_bounds(
+                west, south, east, north, transform=src.transform)
+        if not src.read_masks(1, window=tile_window).astype('bool').any():
+            return None, None
```

Comment:
@jdmcbr thanks for your patience, I finally get it! This code skips unnecessary reprojection. I think it might be more robust if we use `transform_bounds` (https://github.com/mapbox/rasterio/blob/master/rasterio/warp.py#L109) instead of `transform` and perhaps expand the window from mask reading a pixel or two to account for warp edges.

The `.astype('bool')` is superfluous here. We don't need it because the mask has only values of `255` (true) or `0` (false).

Reply:
@sgillies No problem; it had seemed when I first looked it it that avoiding the reprojection was important, but that was based on my intuition and limited testing of components on the command line. Given your intuition in this domain is considerably more developed than mine, I was happy to just support whatever effort would lead to faster processing time on rasters with significant empty space.

Thanks for the comments. Is there a slick way to expand a window by a pixel or two? I don't think I've ever noticed anything like that before, and didn't find anything now in search. But I can also do something not-slick.